src/dmd/target.d

```diff
@@ -1240,6 +1242,67 @@ struct TargetC
             assert(0);
     }
 
+    /**
+     * Builds the C `__builtin_va_list` type for target.
+     * NOTE: This is used for sources compiled in ImportC mode.
+     * Returns:
+     *      `Type` that represents `__builtin_va_list`.
+     */
+    extern (C++) Type va_listType() const
+    {
+        import dmd.mtype : TypePointer;
+
+        if (target.os == Target.OS.Windows)
+        {
+            return new TypePointer(Type.tchar);
+        }
+        else if (target.os & Target.OS.Posix)
+        {
+            if (target.is64bit)
+            {
+                import dmd.arraytypes : Dsymbols;
+                import dmd.declaration: VarDeclaration;
+                import dmd.dstruct : StructDeclaration;
+                import dmd.globals : Loc;
+                import dmd.id : Id;
+                import dmd.identifier : Identifier;
+                import dmd.init: VoidInitializer;
+
+                /* struct __va_list_tag
+                 * {
+                 *    uint, uint, void*, void*
+                 * }
+                 */
+                auto sd = new StructDeclaration(Loc.initial, Id.va_list_tag, false);
+                auto initializer = new VoidInitializer(Loc.initial);
+
+                auto members = new Dsymbols(4);
+                (*members)[0] = new VarDeclaration(Loc.initial, Type.tuns32,
+                                                   Identifier.idPool("gp_offset"), initializer);
+                (*members)[1] = new VarDeclaration(Loc.initial, Type.tuns32,
+                                                   Identifier.idPool("fp_offset"), initializer);
+                (*members)[2] = new VarDeclaration(Loc.initial, Type.tvoidptr,
+                                                   Identifier.idPool("overflow_arg_area"), initializer);
+                (*members)[3] = new VarDeclaration(Loc.initial, Type.tvoidptr,
+                                                   Identifier.idPool("reg_save_area"), initializer);
+                sd.members = members;
```

Comment:
This part is strictly not necessary, but it makes the debugging experience more pleasant.